deployment/ccip/changeset/deployer_group.go

```diff
@@ -94,77 +126,141 @@ func (d *DeployerGroup) GetDeployer(chain uint64) (*bind.TransactOpts, error) {
 		startingNonce = new(big.Int).SetUint64(nonce)
 	}
 
+	dc := d.deploymentContext
 	sim.Signer = func(a common.Address, t *types.Transaction) (*types.Transaction, error) {
+		txCount, err := d.getTransactionCount(chain)
+		if err != nil {
+			return nil, err
+		}
+
+		currentNone := big.NewInt(0).Add(startingNonce, txCount)
+
 		// Update the nonce to consider the transactions that have been sent
-		sim.Nonce = big.NewInt(0).Add(startingNonce, big.NewInt(int64(len(d.transactions[chain]))+1))
+		sim.Nonce = big.NewInt(0).Add(currentNone, big.NewInt(1))
```

Comment:
I think we should move this increment to the end of the function, i.e. after we call oldSigner. It makes it clearer that the nonce is being incremented for the _next_ transaction to be signed, and that the incremented nonce isn't actually being used for _this_ transaction.